editor/components/CrashMapCard.tsx

```diff
@@ -55,71 +68,108 @@ export default function CrashMapCard({
 
   return (
     <Card className="h-100">
-      <Card.Header>
-        Location:{" "}
-        {hasLocation ? (
-          <Link href={`/locations/${locationId}`}>{locationId}</Link>
-        ) : (
-          "unassigned"
-        )}
+      <Card.Header className="d-flex justify-content-between">
+        <div>
+          <span className="fw-bold me-2">Location</span>
+          {hasLocation ? (
+            <Link href={`/locations/${locationId}`}>{locationId}</Link>
+          ) : (
+            <span className="text-secondary">unassigned</span>
+          )}
+        </div>
+        <div>
+          <span className="fw-bold me-2">Provider </span>
+          <span>
+            {" "}
+            {hasCoordinates
+              ? isManualGeocode
+                ? "Manual Q/A"
+                : "TxDOT CRIS"
+              : "No Primary Coordinates"}
+          </span>
+        </div>
       </Card.Header>
       <Card.Body className="p-1 crash-header-card-body" ref={mapContainerRef}>
         <CrashMap
           savedLatitude={savedLatitude}
           savedLongitude={savedLongitude}
-          isEditing={isEditingCoordinates}
-          editCoordinates={editCoordinates}
-          setEditCoordinates={setEditCoordinates}
+          isEditing={isEditing}
+          mapLatLon={mapLatLon}
+          setMapLatLon={setMapLatLon}
           mapRef={mapRef}
         />
       </Card.Body>
       <Card.Footer>
-        <div className="d-flex justify-content-between">
-          <div>
-            <span>
-              Location provider: {""}
-              {hasCoordinates
-                ? isManualGeocode
-                  ? "Manual Q/A"
-                  : "TxDOT CRIS"
-                : "No Primary Coordinates"}
-            </span>
-          </div>
-          <PermissionsRequired allowedRoles={allowedMapEditRoles}>
-            <div>
+        <PermissionsRequired allowedRoles={allowedMapEditRoles}>
+          <div
+            className={`d-flex align-items-center ${isEditing ? "justify-content-between" : "justify-content-end"}`}
+          >
+            {isEditing && (
+              <div className="flex-grow-1">
+                <CrashMapCoordinateForm
+                  mapLatLon={mapLatLon}
+                  formLatLon={formLatLon}
+                  setFormLatLon={setFormLatLon}
+                  validationError={validationError}
+                  setValidationError={setValidationError}
+                />
+              </div>
+            )}
+            <div className="d-flex text-nowrap d-flex align-items-center justify-content-end">
               <Button
                 size="sm"
                 variant="primary"
                 disabled={isMutating}
                 onClick={async () => {
-                  if (!isEditingCoordinates) {
-                    setIsEditingCoordinates(true);
+                  if (!isEditing) {
+                    setIsEditing(true);
                   } else {
```

Comment:
when we save, we check if the map coordinates match the form coordinates, if not, we validate the form coordinates and use them if they var valid